src/Components/HomePage/APIcall.js

```diff
@@ -0,0 +1,23 @@
+const APIcall = async (url='', dataSetter, errorSetter, loaderSetter) => {
+    try{
+        const response = await fetch(url);
+
+        if (!response.ok) {
+            throw new Error("HTTP error " + response.status);
+        }
+        const data = await response.json();
+
+        data.results.length > 0 ?
+            dataSetter(data.results)
+        :   console.log('There are no results', data);
+
+        console.log(data.results);
+
+    } catch (error){
+        errorSetter(error)
+    } finally{
+        loaderSetter(false)
```

Comment:
Perhaps instead of having these handlers you could provide a `callback` function, remember to keep the principle of single responsability. The API Call function shouldn't need to know what to do with the data, norr how to handle errors.

Those `console.logs` will create noise in the console and might cause you troubles while debugging. Avoid leaving `console.logs` behind, as you may accidentally leak keys over the user's console.